calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -195,11 +195,13 @@ func (rh resourceHelper) Update(ctx context.Context, client client.Interface, re
 	// Check to see if the resourceVersion is specified in the resource object.
 	rv := resource.(ResourceObject).GetObjectMeta().GetResourceVersion()
 
+	// Copy the resource to prevent modifying the input resource metadata.
+	ru := resource.DeepCopyObject().(ResourceObject)
```

Comment:
actually I would have done `resource =` which I think is safer since it means you won't be able to accidentally beat over the original.